shell/platform/common/cpp/text_input_model.h

```diff
@@ -9,7 +9,55 @@
 #include <memory>
 #include <string>
 
+#include "flutter/fml/logging.h"
+
 namespace flutter {
+
+// A directional range of text.
+//
+// A |TextRange| describes a range of text with |base| and |extent| positions.
+// In the case where |base| == |extent|, the range is said to be collapsed, and
+// when |base| > |extent|, the range is said to be reversed.
+class TextRange {
```

Comment:
Any reason not to pull this out into a separate file? (Same then for tests.)

Reply:
Done.